File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/ElementEquals.java
package io.appium.espressoserver.lib.handlers;

import android.view.View;

import javax.annotation.Nullable;

import io.appium.espressoserver.lib.handlers.exceptions.AppiumException;
import io.appium.espressoserver.lib.model.AppiumParams;
import io.appium.espressoserver.lib.model.Element;

public class ElementEquals implements RequestHandler<AppiumParams, Boolean> {

    @Override
    @Nullable
    public Boolean handle(AppiumParams params) throws AppiumException {
        final String elementId = params.getElementId();
        final String otherElementId = params.getUriParameterValue("otherId");
        View viewOne = Element.getViewById(elementId);
        View viewTwo = Element.getViewById(otherElementId);
        return viewOne.equals(viewTwo);
    }
}

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/ElementEquals.kt
package io.appium.espressoserver.lib.handlers

import io.appium.espressoserver.lib.handlers.exceptions.AppiumException
import io.appium.espressoserver.lib.handlers.exceptions.InvalidArgumentException
import io.appium.espressoserver.lib.model.AppiumParams
import io.appium.espressoserver.lib.model.EspressoElement

class ElementEquals : RequestHandler<AppiumParams, Boolean> {

    @Throws(AppiumException::class)
    override fun handleInternal(params: AppiumParams): Boolean {
        val elementId = params.elementId
        val otherElementId = params.getUriParameterValue("otherId")
                ?: throw InvalidArgumentException("'otherElementId' query parameter not found")
        val viewOne = EspressoElement.getCachedViewStateById(elementId).view
        val viewTwo = EspressoElement.getCachedViewStateById(otherElementId).view
        return viewOne == viewTwo
    }
}